### ISSUE TITLE
undefined method `clang_version' for OS::Mac:Module

### ISSUE THREAD

**Issue by rolandjitsu** [Homebrew/brew#219]
It looks like any commands I run (`brew cleanup`, `brew install <package>`, etc.) keeps throwing the same error:
```
Error: undefined method `clang_version' for OS::Mac:Module
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Taps/homebrew/homebrew-science/octave.rb:32:in `<class:Octave>'
/usr/local/Library/Taps/homebrew/homebrew-science/octave.rb:1:in `load_formula'
/usr/local/Library/Homebrew/formulary.rb:21:in `module_eval'
/usr/local/Library/Homebrew/formulary.rb:21:in `load_formula'
/usr/local/Library/Homebrew/formulary.rb:38:in `load_formula_from_path'
/usr/local/Library/Homebrew/formulary.rb:87:in `load_file'
/usr/local/Library/Homebrew/formulary.rb:78:in `klass'
/usr/local/Library/Homebrew/formulary.rb:74:in `get_formula'
/usr/local/Library/Homebrew/formulary.rb:211:in `factory'
/usr/local/Library/Homebrew/formulary.rb:220:in `from_rack'
/usr/local/Library/Homebrew/cleanup.rb:80:in `block in cleanup_cache'
/usr/local/Library/Homebrew/cleanup.rb:48:in `each'
/usr/local/Library/Homebrew/cleanup.rb:48:in `cleanup_cache'
/usr/local/Library/Homebrew/cleanup.rb:11:in `cleanup'
/usr/local/Library/Homebrew/cmd/cleanup.rb:20:in `cleanup'
/usr/local/Library/brew.rb:87:in `<main>'
```

This started happening after I installed octave from [hombrew-science](https://github.com/Homebrew/homebrew-science).

I did try to update homebrew and upgrade all formulae, but I keep getting the same message. Though, running `brew doctor` seems to not throw that error.

**Comment by glaszig** [comment]
just did a `brew update` and now get a similar issue. related?

```
$ brew upgrade imagemagick
==> Upgrading 1 outdated package, with result:
imagemagick 6.9.3-10
==> Upgrading imagemagick
==> Installing dependencies for imagemagick: libtiff, freetype, fontconfig
==> Installing imagemagick dependency: libtiff
==> Downloading https://homebrew.bintray.com/bottles/libtiff-4.0.6.el_capitan.bottle.tar.gz
Already downloaded: /Library/Caches/Homebrew/libtiff-4.0.6.el_capitan.bottle.tar.gz
==> Pouring libtiff-4.0.6.el_capitan.bottle.tar.gz
Error: undefined method `default_compiler' for OS::Mac:Module
Warning: Bottle installation failed: building from source.
==> Downloading http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz
Already downloaded: /Library/Caches/Homebrew/libtiff-4.0.6.tar.gz
==> ./configure --prefix=/usr/local/Cellar/libtiff/4.0.6 --without-x --disable-lzma --with-jpeg-include-dir=/usr/local/opt/jpeg/include --with-jpeg-lib-dir=/usr/local/opt/jpeg/lib
==> make install
Error: undefined method `default_compiler' for OS::Mac:Module
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/tab.rb:190:in `compiler'
/usr/local/Library/Homebrew/tab.rb:196:in `cxxstdlib'
/usr/local/Library/Homebrew/cxxstdlib.rb:61:in `block in check_dependencies'
/usr/local/Library/Homebrew/cxxstdlib.rb:56:in `each'
/usr/local/Library/Homebrew/cxxstdlib.rb:56:in `check_dependencies'
/usr/local/Library/Homebrew/cxxstdlib.rb:29:in `check_compatibility'
/usr/local/Library/Homebrew/build.rb:145:in `detect_stdlibs'
/usr/local/Library/Homebrew/build.rb:133:in `block in install'
/usr/local/Library/Homebrew/formula.rb:934:in `block in brew'
/usr/local/Library/Homebrew/formula.rb:1563:in `block in stage'
/usr/local/Library/Homebrew/resource.rb:98:in `block in unpack'
/usr/local/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/usr/local/Library/Homebrew/extend/fileutils.rb:75:in `block in run'
/usr/local/Library/Homebrew/extend/fileutils.rb:75:in `chdir'
/usr/local/Library/Homebrew/extend/fileutils.rb:75:in `run'
/usr/local/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/usr/local/Library/Homebrew/resource.rb:94:in `unpack'
/usr/local/Library/Homebrew/resource.rb:86:in `stage'
/usr/local/Library/Homebrew/formula.rb:1553:in `stage'
/usr/local/Library/Homebrew/formula.rb:929:in `brew'
/usr/local/Library/Homebrew/build.rb:108:in `install'
/usr/local/Library/Homebrew/build.rb:178:in `<main>'
```

**Comment by UniqMartin** [comment]
@glaszig Yes, it's related but different. Should be resolved by 86fa42b36c98e482490ca347849e041ee9316011. Can you `brew update` and retry?

@rolandjitsu This issue seems a bit more involved and is related to changes made today in #168. I'm looking into this, but tagging the original author (@mikemcquaid) so that he will notice and possibly beat me to resolving this.

**Comment by glaszig** [comment]
@UniqMartin works for me. thx.
